### PR TITLE
tag: Add --version-id support

### PR DESCRIPTION
**Pull request by vadmeste**
mc tag list/set/remove supports --version-id now.

### CODE REVIEW

**Comment by vadmeste** [comment]
@kannappanr all comments addressed here

**Comment by kannappanr** [comment]
looks OK except the `version-id and vid` flags are aliases, like in

https://github.com/minio/mc/blob/master/cmd/cat-main.go#L43
https://github.com/minio/mc/blob/master/cmd/cp-main.go#L45
https://github.com/minio/mc/blob/master/cmd/rm-main.go#L47

File: cmd/tag-list.go [hunk 1]
```diff
@@ -53,22 +69,29 @@ EXAMPLES:
   1. List the tags assigned to an object.
      {{.Prompt}} {{.HelpName}} myminio/testbucket/testobject
 
-  2. List the tags assigned to an object in JSON format.
+  2. List the tags assigned to particular version of an object.
+     {{.Prompt}} {{.HelpName}} --vid "ieQq7aXsyhlhDt47YURGlrucYY3GxWHa" myminio/testbucket/testobject
+
+  3. List the tags assigned to an object versions that are older than one week.
```
Comment (poornas) [inline cmd/tag-list.go:75]:
```suggestion
  3. List the tags assigned to an object's versions that are older than one week.
```

File: cmd/tag-list.go [hunk 2]
```diff
@@ -53,22 +69,29 @@ EXAMPLES:
...
 
 // tagListMessage structure for displaying tag
 type tagListMessage struct {
-	Tags   map[string]string `json:"tagset,omitempty"`
-	Status string            `json:"status"`
-	URL    string            `json:"url"`
+	Tags      map[string]string `json:"tagset,omitempty"`
+	Status    string            `json:"status"`
+	URL       string            `json:"url"`
+	VersionID string            `json:"versionID"`
```
Comment (poornas) [inline cmd/tag-list.go:94]:
```suggestion
	VersionID string            `json:"versionID,omitempty"`
```

File: cmd/tag-remove.go [hunk 3]
```diff
@@ -48,20 +64,33 @@ EXAMPLES:
...
 `,
 }
 
 // tagSetTagMessage structure will show message depending on the type of console.
 type tagRemoveMessage struct {
-	Status string `json:"status"`
-	Name   string `json:"name"`
+	Status    string `json:"status"`
+	Name      string `json:"name"`
+	VersionID string `json:"versionID"`
```
Comment (poornas) [inline cmd/tag-remove.go:82]:
```suggestion
	VersionID string `json:"versionID,omitempty"`
```

File: cmd/tag-set.go [hunk 4]
```diff
@@ -48,20 +62,33 @@ EXAMPLES:
   1. Assign tags to an object.
      {{.Prompt}} {{.HelpName}} play/testbucket/testobject "key1=value1&key2=value2&key3=value3"
 
-  2. Assign tags to a bucket.
+  2. Assign tags to a particuler version of an object.
```
Comment (poornas) [inline cmd/tag-set.go:65]:
```suggestion
  2. Assign tags to a particular version of an object.
```

File: cmd/tag-set.go [hunk 5]
```diff
@@ -48,20 +62,33 @@ EXAMPLES:
...
+     {{.Prompt}} {{.HelpName}} --vid "ieQq7aXsyhlhDt47YURGlrucYY3GxWHa" play/testbucket/testobject "key1=value1&key2=value2&key3=value3"
+
+  3. Assign tags to a object versions older than one week.
```
Comment (poornas) [inline cmd/tag-set.go:68]:
```suggestion
  3. Assign tags to all object versions older than a week.
```

File: cmd/tag-set.go [hunk 6]
```diff
@@ -48,20 +62,33 @@ EXAMPLES:
...
 `,
 }
 
 // tagSetTagMessage structure will show message depending on the type of console.
 type tagSetMessage struct {
-	Status string `json:"status"`
-	Name   string `json:"name"`
+	Status    string `json:"status"`
+	Name      string `json:"name"`
+	VersionID string `json:"versionID"`
```
Comment (poornas) [inline cmd/tag-set.go:80]:
```suggestion
	VersionID string `json:"versionID,omitempty"`
```

File: cmd/tag-set.go [hunk 7]
```diff
@@ -71,30 +98,69 @@ func (t tagSetMessage) JSON() string {
...
 	}
+
+	targetURL = ctx.Args().Get(0)
+	tags = ctx.Args().Get(1)
+	versionID = ctx.String("vid")
+	withVersions = ctx.Bool("versions")
+	rewind := ctx.String("rewind")
+
+	if versionID != "" && (rewind != "" || withVersions) {
+		fatalIf(errDummy().Trace(), "You cannot specify both --vid and --rewind or --versions flags at the same time")
```
Comment (poornas) [inline cmd/tag-set.go:113]:
```suggestion
		fatalIf(errDummy().Trace(), "You cannot specify --vid and --rewind or --versions flags at the same time")
```

**Comment by vadmeste** [comment]
> looks OK except the `version-id and vid` flags are aliases, like in
> 
> https://github.com/minio/mc/blob/master/cmd/cat-main.go#L43
> https://github.com/minio/mc/blob/master/cmd/cp-main.go#L45
> https://github.com/minio/mc/blob/master/cmd/rm-main.go#L47

oh yes, done now.

**Pull request merged**